doc/sphinx/user-extensions/proof-schemes.rst

```diff
@@ -8,35 +8,36 @@ Proof schemes
 Generation of induction principles with ``Scheme``
 --------------------------------------------------------
 
-The ``Scheme`` command is a high-level tool for generating automatically
-(possibly mutual) induction principles for given types and sorts. Its
-syntax follows the schema:
+.. cmd:: Scheme {? @ident := } @scheme_kind {* with {? @ident := } @scheme_kind }
 
-.. cmd:: Scheme @ident__1 := Induction for @ident__2 Sort @sort {* with @ident__i := Induction for @ident__j Sort @sort}
+   .. insertprodn scheme_kind sort_family
 
-  This command is a high-level tool for generating automatically
+   .. prodn::
+      scheme_kind ::= Equality for @reference
+      | {| Induction | Minimality | Elimination | Case } for @reference Sort @sort_family
+      sort_family ::= Set
+      | Prop
+      | SProp
+      | Type
+
+  A high-level tool for automatically generating
   (possibly mutual) induction principles for given types and sorts.
-  Each :n:`@ident__j` is a different inductive type identifier belonging to
+  Each :n:`@ident` is a different inductive type identifier belonging to
   the same package of mutual inductive definitions.
-  The command generates the :n:`@ident__i`\s to be mutually recursive
-  definitions. Each term :n:`@ident__i` proves a general principle of mutual
-  induction for objects in type :n:`@ident__j`.
-
-.. cmdv:: Scheme @ident := Minimality for @ident Sort @sort {* with @ident := Minimality for @ident' Sort @sort}
-
-   Same as before but defines a non-dependent elimination principle more
-   natural in case of inductively defined relations.
-
-.. cmdv:: Scheme Equality for @ident
-   :name: Scheme Equality
+  The command generates the :n:`@ident`\s as mutually recursive
+  definitions. Each term :n:`@ident` proves a general principle of mutual
+  induction for objects in type :n:`@reference`.
 
-   Tries to generate a Boolean equality and a proof of the decidability of the usual equality. If `ident`
-   involves some other inductive types, their equality has to be defined first.
+  :n:`@reference`
+    The name of the scheme. If not provided, the scheme name will be determined automatically
+    from the sorts involved.
 
-.. cmdv:: Scheme Induction for @ident Sort @sort {* with Induction for @ident Sort @sort}
+  :n:`Minimality`
```

Comment:
For consistency with the presentation of the `Equality` clause below, this should be:
```suggestion
  :n:`Minimality for @reference Sort @sort_family`
```